README.md

```diff
@@ -1,5 +1,92 @@
 # Argo Rollout Gloo Edge Plugin
 
-An Argo Rollouts plugin for [Gloo Edge](https://www.solo.io/products/gloo-edge/).
+An Argo Rollouts plugin for [Gloo Edge](https://www.solo.io/products/gloo-edge/). Documentation for Argo Rollouts that covers canary rollouts can be found [here](https://argo-rollouts.readthedocs.io/en/stable/features/canary/#canary-deployment-strategy).
 
-Current state of main branch is initial commit - nothing actually working yet.
+## VirtualService based Canary Rollouts
+
+A snippet of of a rollout configuration that contains Gloo Edge plugin configuration for VirtualService-based rollouts:
+```
+  strategy:
+    canary:
+      canaryService: echo-v2 
+      stableService: echo-v1
+      trafficRouting:
+        plugins:
+          solo-io/glooedge:
+            virtualService:
+              name: echo
+              namespace: gloo-system
+            routes:
+              - route-1
+              - route-2
+      steps:
+        - setWeight: 10
+        - pause: {}
+```
+
+`stableService` field contains the name of the k8s Service pointing to a stable release. `canaryService` field contains the name of the service pointing to a canary release.
+
+Everything under `plugins:solo-io/glooedge` is a configuration for the Gloo Edge plugin. `virtualService` specifies which VirtualService is used to route traffic to stable and canary versions of the service. The plugin will update weights in the selected VirtualService according to rollout steps in the rollout configuration. 
+
+If there are multiple routes present in the VirtualHost of a VirtualService, the routes where weights will need to be updated during the rollout must be listed under `routes`. Otherwise this setting is optional. All routes listed under `routes` must exist.
+
+An example VirtualService configuration:
+```
+apiVersion: gateway.solo.io/v1
+kind: VirtualService
+metadata:
+  name: echo
+  namespace: gloo-system
+spec:
+  virtualHost:
+    domains:
+    - '*'
+    routes:
+    - matchers:
+      - prefix: /
+      routeAction:
+        multi:
+          destinations:
+          - destination:
+              upstream:
+                name: echo-v1
+                namespace: echo
+            weight: 100
+```
+
+Note that only `multi` routeActions are supported. It's ok to define a destination for a stable release only. The names for stable and canary `Upstream`s are expected to match the name of the services (and `stableService` and `canaryService` fields of the plugin configuration).
```

Comment:
You added support for single destinations also right?

Reply:
Yes, see here: https://github.com/argoproj-labs/rollouts-plugin-trafficrouter-glooedge/pull/1/files#diff-56d28ecf3c4e8c775d9ec794e963fde93acc1b78444376ebb8a8a9a7a9b0c415R30 and here: https://github.com/argoproj-labs/rollouts-plugin-trafficrouter-glooedge/pull/1/files#diff-ab569cd7f37c9fd5ca9dfdeda525bc8fd68af3442955c6165e067321e1a942fbR33